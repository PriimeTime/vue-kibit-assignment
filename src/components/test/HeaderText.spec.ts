import { describe, it, expect } from "vitest";
import { mount } from "@vue/test-utils";
import HeaderText from "@/components/HeaderText.vue";

describe("HeaderText", () => {
  it("renders without crashing", () => {
    const wrapper = mount(HeaderText);
    expect(wrapper.exists()).toBe(true);
  });

  it("renders slot content", () => {
    const slotContent = "Hello, World!";
    const wrapper = mount(HeaderText, {
      slots: {
        default: slotContent,
      },
    });
    expect(wrapper.text()).toContain(slotContent);
  });

  it("matches snapshot", () => {
    const wrapper = mount(HeaderText, {
      slots: {
        default: "Snapshot Header",
      },
    });
    expect(wrapper.html()).toMatchSnapshot();
  });
});
