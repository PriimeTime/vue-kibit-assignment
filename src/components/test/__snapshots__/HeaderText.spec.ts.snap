// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`HeaderText > matches snapshot 1`] = `"<div class="text-3xl font-bold">Snapshot Header</div>"`;
